contracts/Voter.sol

```diff
@@ -1,88 +1,100 @@
+import "contracts/interfaces/IGauge.sol";
+import "contracts/interfaces/IGaugeFactory.sol";
+import "contracts/interfaces/IERC20.sol";
+import "contracts/interfaces/IMinter.sol";
+import "contracts/interfaces/IPair.sol";
+import "contracts/interfaces/IPairFactory.sol";
+import "contracts/interfaces/IVoter.sol";
+import "contracts/interfaces/IVotingEscrow.sol";
+
+import "contracts/Pair.sol";
+
+import "contracts/interfaces/IWrappedExternalBribeFactory.sol";
 
 contract Voter is IVoter {
-
     address public immutable _ve; // the ve token that governs these contracts
     address public immutable factory; // the PairFactory
     address internal immutable base;
     address public immutable gaugefactory;
     address public immutable bribefactory;
-    uint internal constant DURATION = 7 days; // rewards are released over 7 days
+    address public immutable wrappedxbribefactory; // this is the address of the wrapped external bribe factory
+    uint256 internal constant DURATION = 7 days; // rewards are released over 7 days
     address public minter;
     address public governor; // should be set to an IGovernor
     address public emergencyCouncil; // credibly neutral party similar to Curve's Emergency DAO
-
-    uint public totalWeight; // total voting weight
+    uint256 public totalWeight; // total voting weight
 
     address[] public pools; // all pools viable for incentives
     mapping(address => address) public gauges; // pool => gauge
     mapping(address => address) public poolForGauge; // gauge => pool
     mapping(address => address) public internal_bribes; // gauge => internal bribe (only fees)
     mapping(address => address) public external_bribes; // gauge => external bribe (real bribes)
     mapping(address => uint256) public weights; // pool => weight
-    mapping(uint => mapping(address => uint256)) public votes; // nft => pool => votes
-    mapping(uint => address[]) public poolVote; // nft => pools
-    mapping(uint => uint) public usedWeights;  // nft => total voting weight of user
-    mapping(uint => uint) public lastVoted; // nft => timestamp of last vote, to ensure one vote per epoch
+    mapping(uint256 => mapping(address => uint256)) public votes; // nft => pool => votes
+    mapping(uint256 => address[]) public poolVote; // nft => pools
+    mapping(uint256 => uint256) public usedWeights; // nft => total voting weight of user
+    mapping(uint256 => uint256) public lastVoted; // nft => timestamp of last vote, to ensure one vote per epoch
     mapping(address => bool) public isGauge;
     mapping(address => bool) public isWhitelisted;
     mapping(address => bool) public isAlive;
 
-    event GaugeCreated(address indexed gauge, address creator, address internal_bribe, address indexed external_bribe, address indexed pool);
+    event GaugeCreated(
+        address indexed gauge,
+        address creator,
+        address internal_bribe,
+        address indexed external_bribe,
+        address wxbribe,
```

Comment:
Suggest not to change this event but put wrapped external bribe contract address in `external_bribe`